packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -376,7 +367,7 @@ library LibProving {
     // 6.5625.
     function _overrideWithHigherProof(
         TaikoData.Block storage _blk,
-        TaikoData.TransitionState storage _ts,
+        TaikoData.TransitionState memory _ts,
```

Comment:
I'm not sure this works !

In this function we shall persist some data on-chain - but obviously with `memory` we dont:

```
   _ts.validityBond = _local.tier.validityBond;
        _ts.contester = address(0);
        _ts.prover = msg.sender;
        _ts.tier = _proof.tier;

        if (!_local.sameTransition) {
            _ts.blockHash = _tran.blockHash;
            _ts.stateRoot = _tran.stateRoot;
        }
```

Reply:
Ok, i think this is what made me sceptical:
![kép](https://github.com/taikoxyz/taiko-mono/assets/51912515/3070b522-bc7a-4827-8e9e-f8fca2b8ba5a)

But i verified that `memoy struct` is passed by value to private functions, so so far seems good !